superset/sql_lab.py

```diff
@@ -272,7 +272,10 @@ def handle_error(msg):
         key = '{}'.format(uuid.uuid4())
         logging.info('Storing results in results backend, key: {}'.format(key))
         json_payload = json.dumps(payload, default=utils.json_iso_dttm_ser)
-        results_backend.set(key, utils.zlib_compress(json_payload))
+        cache_timeout = database.cache_timeout
+        if cache_timeout is None:
+            cache_timeout = config.get('CACHE_DEFAULT_TIMEOUT', 0)
+        results_backend.set(key, utils.zlib_compress(json_payload), cache_timeout)
```

Comment:
Oh interesting. FYI we use the s3 caching as a result backend which doesn't really support timeout, so this won't matter on our end, but for people using other results backend that's probably the desired behavior. https://github.com/bkyryliuk/s3werkzeugcache/blob/master/s3cache/s3cache.py